rules/implicit_aliasing.go

```diff
@@ -0,0 +1,92 @@
+package rules
+
+import (
+	"fmt"
+	"github.com/securego/gosec"
+	"go/ast"
+)
+
+type implicitAliasing struct {
+	gosec.MetaData
+	maxDepth int
+}
+
+func (r *implicitAliasing) ID() string {
+	return r.MetaData.ID
+}
+
+func isDereferenceOf(n ast.Node, name string) bool {
+	if unary, ok := n.(*ast.UnaryExpr); ok && unary.Op.String() == "&" {
+		if ident, ok := unary.X.(*ast.Ident); ok && ident.Name == name {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *implicitAliasing) Match(n ast.Node, c *gosec.Context) (*gosec.Issue, error) {
+	if rangeStmt, ok := n.(*ast.RangeStmt); ok && rangeStmt.Value != nil && rangeStmt.Tok.String() == ":=" {
+		if ident, ok := rangeStmt.Value.(*ast.Ident); ok {
+			name := ident.Name
+
+			var issue *gosec.Issue
+			var fun func(n ast.Node) bool
+			fun = func(n ast.Node) bool {
+				if ret, ok := n.(*ast.ReturnStmt); ok {
+					for _, element := range ret.Results {
+						if isDereferenceOf(element, name) {
+							continue
+						}
+
+						ast.Inspect(element, fun)
```

Comment:
We try to avoid `Inspect` in the rules, because will effective traverse again the tree which is not very efficient. Maybe have a look at this rule which uses a cache to traverse only once the tree https://github.com/securego/gosec/blob/a305f10eb98e67d7fd0259d369ded9d3caf31ded/rules/integer_overflow.go#L33.